bin/ch/WScriptJsrt.cpp

```diff
@@ -426,6 +426,134 @@ JsValueRef WScriptJsrt::ClearTimeoutCallback(JsValueRef callee, bool isConstruct
     return JS_INVALID_REFERENCE;
 }
 
+template <class DebugOperationFunc>
+void QueueDebugOperation(JsValueRef function, const DebugOperationFunc& operation)
+{
+    WScriptJsrt::PushMessage(WScriptJsrt::CallbackMessage::Create(function, operation));
+}
+
+JsValueRef WScriptJsrt::AttachCallback(JsValueRef callee, bool isConstructCall, JsValueRef *arguments, unsigned short argumentCount, void *callbackState)
+{
+    LPWSTR errorMessage = L"WScript.Attach requires a function, like WScript.Attach(foo);";
+    if (argumentCount != 2)
+    {
+        goto Error;
+    }
+    JsValueType argumentType = JsUndefined;
+    IfJsrtError(ChakraRTInterface::JsGetValueType(arguments[1], &argumentType));
+    if (argumentType != JsFunction)
+    {
+        goto Error;
```

Comment:
error goes to javascript so that if it have a try catch it can be handled

---
In reply to: [62719870](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62719870) [](ancestors = 62719870)